src/common/models/dimension/dimensions.ts

```diff
@@ -132,4 +132,8 @@ export class Dimensions {
   toJS(): DimensionOrGroupJS[] {
     return this.dimensions.map(dimensionOrGroup => dimensionOrGroup.toJS());
   }
+
+  slice(begin: number, end?: number): Dimensions {
```

Comment:
Do we need `Dimensions` class? Constructor looks costly, maybe we could just slice `flattenedDimensions` and return `Dimension[]`?

Reply:
It was not used at all 😂